### PR TITLE
make nb-repository-plugin download runnable on  maven 3.5+

### PR DESCRIPTION
**Pull request by ebarboni**
Hi, this PR is to try to make nb-repository compatible with 3.5.0 maven

due to https://issues.apache.org/jira/browse/MINDEXER-106, I try to upgrade indexer to 6.0-SNAPSHOT to experiment
It require java7 compatibility upgrade to 5.5.4 lucene and a few changes in populatemojo.
Doing this changes make the plugin non compatible with 3.0.5 maven version. 

Assuming indexer 6.0 is out:
Do we still need to stick with the old maven ? Should we make 2 branches ?

Regards

### CODE REVIEW

**Comment by mkleint** [comment]
if the only incompatibility with 3.0.x is jdk6->jdk7 move, I'm happy to just ignore that. noone should be using jdk 1.6 anymore (neither 1.7)

I'm assuming that the lucene and maven indexer dependencies are private to the plugin and don't cause any issues at runtime, is that right?

**Comment by ebarboni** [comment]
I'm not sure of the dependencies scope.
#4 runtime is broken on maven 3.0.5 because injection of some class is not working
and without #4 runtime on maven 3.5.2 is broken because a class is not found.

**Comment by ebarboni** [comment]
Hi, updated to release indexer 6.0.0. Seems ok Prerequisite to 3.1.1 to prevent usage from incompatible former maven.

**Comment by ebarboni** [comment]
@mkleint, do you think this can be accepted ?

File: pom.xml [hunk 1]
```diff
@@ -340,23 +383,23 @@
...
             <groupId>org.apache.maven</groupId>
             <artifactId>maven-artifact</artifactId>
-            <version>3.0.5</version>
+            <version>3.5.2</version>
         </dependency>
         <dependency>
             <groupId>org.apache.maven</groupId>
             <artifactId>maven-plugin-api</artifactId>
-            <version>3.0.5</version>
+            <version>3.5.2</version>
```
Comment (mkleint) [inline pom.xml]: we have 3.5.2 dependency here but above we only declare 3.1.1 as maven version?

Comment (ebarboni) [reply pom.xml]: adjusted to 3.1.1 for now. Not sure how connected prerequiste and maven version must be linked. But for now it is not  mandatory to jump for last version

**Pull request merged**